server/src/lib.rs

```diff
@@ -1424,24 +1389,41 @@ mod tests {
             .expect("failed to create database");
         let bananas_uuid = bananas.uuid().unwrap();
 
-        std::mem::drop(server);
+        assert!(bananas.is_initialized());
+
+        // Shutdown server
+        server.shutdown();
+        server.join().await.unwrap();
+
+        assert!(bananas.is_shutdown());
 
         let server = make_server(Arc::clone(&application));
         server.set_id(ServerId::try_from(1).unwrap()).unwrap();
         server.wait_for_init().await.unwrap();
 
+        let bananas = server.database(&bananas_name).unwrap();
+        bananas.wait_for_init().await.unwrap();
+
         let apples = create_simple_database(&server, "apples")
             .await
             .expect("failed to create database");
+        assert!(apples.is_initialized());
+
         let apples_uuid = apples.uuid().unwrap();
 
         assert_eq!(server.db_names_sorted(), vec!["apples", "bananas"]);
 
-        std::mem::drop(server);
+        let bananas_object_store = bananas.iox_object_store().unwrap();
 
-        bananas
-            .iox_object_store()
-            .unwrap()
+        // Shutdown server
```

Comment:
I don't know if it would add any value but you could also add:

```suggestion
        assert!(!apples.is_shutdown());
        assert!(!bananas.is_shutdown());
        // Shutdown server
```

To demonstrate that the server shutdown causes the databases to shutdown